### PR TITLE
javascript-carousel - James Ding

### PR DESCRIPTION
**Pull request by jamessding**
Kapture 2022-06-08 at 16 41 37

### CODE REVIEW

**Comment by jamessding** [comment]
Kapture 2022-06-09 at 01 07 30

File: javascript-carousel/main.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      $image.setAttribute('data-id', 0);
+      setTimeout(function () { $image.setAttribute('src', urlArray[0]); }, 100);
+      $dots[0].className = 'fas fa-circle fa-2x padding';
+      $dots[(urlArrayLength - 1).toString()].className = 'far fa-circle fa-2x padding';
+    } else {
+      $image.setAttribute('data-id', Number(imageId) + 1);
+      setTimeout(function () { $image.setAttribute('src', urlArray[Number(imageId) + 1]); }, 100);
+      $dots[Number(imageId) + 1].className = 'fas fa-circle fa-2x padding';
+      $dots[Number(imageId)].className = 'far fa-circle fa-2x padding';
+    }
```
Comment (codingcodymiller) [inline javascript-carousel/main.js:37]: This seems to have a lot of repeated code with the only difference being which index the element is that it is affecting. You should make a function with a parameter for the index to affect and just call that function.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Hey James! In addition to my other feedback, you seem to have quite a few anonymous function definitions here. I suggest switching to named functions.

**Review by codingcodymiller — APPROVED** [review]
Excellent work!!

**Pull request merged**